modules/boss.lua

```diff
@@ -69,13 +120,149 @@ mod.optionsArgs = {
+    name = Encounters:GetInstance(469).name,
+    order = 7,
+    set = SetEP,
+    get = GetEP,
+    args = {
+      razorgore = EncounterPlate(610, 1),
+      vaelastrasz = EncounterPlate(611, 2),
+      broodlord = EncounterPlate(612, 3),
+      firemaw = EncounterPlate(613, 4),
+      ebonroc = EncounterPlate(614, 5),
+      flamegor = EncounterPlate(615, 6),
+      chromaggus = EncounterPlate(616, 7),
+      nefarian = EncounterPlate(617, 8),
+    }
+  },
+  bossrewards_zg = {
+    type = "group",
+    name = Encounters:GetInstance(309).name,
+    order = 8,
+    set = SetEP,
+    get = GetEP,
+    args = {
+      venoxis = EncounterPlate(784, 1),
+      jeklik = EncounterPlate(785, 2),
+      marli = EncounterPlate(786, 3),
+      mandokir = EncounterPlate(787, 4),
+      hazzarah = EncounterPlate(788, 5),
+      thekal = EncounterPlate(789, 6),
+      gahzranka = EncounterPlate(790, 7),
+      arlokk = EncounterPlate(791, 8),
+      jindo = EncounterPlate(792, 9),
+      hakkar = EncounterPlate(793, 10),
+    }
+  },
+  bossrewards_aq20 = {
+    type = "group",
+    name = Encounters:GetInstance(509).name,
+    order = 9,
+    set = SetEP,
+    get = GetEP,
+    args = {
+      kurinnaxx = EncounterPlate(718, 1),
+      rajaxx = EncounterPlate(719, 2),
+      moam = EncounterPlate(720, 3),
+      buru = EncounterPlate(721, 4),
+      ayamiss = EncounterPlate(722, 5),
+      ossirian = EncounterPlate(723, 6),
+    }
+  },
+  bossrewards_aq40 = {
+    type = "group",
+    name = Encounters:GetInstance(531).name,
+    order = 10,
+    set = SetEP,
+    get = GetEP,
+    args = {
+      skeram = EncounterPlate(709, 1),
+      bugtrio = EncounterPlate(710, 2),
+      sartura = EncounterPlate(711, 3),
+      fankriss = EncounterPlate(712, 4),
+      viscidus = EncounterPlate(713, 5),
+      huhuran = EncounterPlate(714, 6),
+      twins = EncounterPlate(715, 7),
+      ouro = EncounterPlate(716, 8),
+      cthun = EncounterPlate(717, 9),
+    }
+  },
+  bossrewards_naxx = {
+    type = "group",
+    name = Encounters:GetInstance(533).name,
+    order = 11,
+    set = SetEP,
+    get = GetEP,
+    args = {
+      anub = EncounterPlate(1107, 1),
+      faerlina = EncounterPlate(1110, 2),
+      maexxna = EncounterPlate(1116, 3),
+      noth = EncounterPlate(1117, 4),
+      heigan = EncounterPlate(1112, 5),
+      loatheb = EncounterPlate(1115, 6),
+      razuvious = EncounterPlate(1113, 7),
+      gothik = EncounterPlate(1109, 8),
+      fourhorse = EncounterPlate(1121, 9),
+      patchwerk = EncounterPlate(1118, 10),
+      grobbulus = EncounterPlate(1111, 11),
+      gluth = EncounterPlate(1108, 12),
+      thaddius = EncounterPlate(1120, 13),
+      sapphiron = EncounterPlate(1119, 14),
+      kelthuzad = EncounterPlate(1114, 15),
+    }
+  }
 }
 
 local function dbmCallback(event, mod)
   Debug("dbmCallback: %s %s", event, mod.combatInfo.name)
-  BossAttempt(event, mod.combatInfo.name)
+  EncounterAttempt(event, mod.combatInfo.eId)
```

Comment:
A simple question: how to guarantee the pair of `mod.combatInfo.name` and `mod.combatInfo.eId` from `DBM` will match the data in `LibEncounters`? Is this library designed for DBM?
``` lua
-- DBM code
local mod	= DBM:NewMod("Kel'Thuzad", "DBM-Naxx", 5)
local L		= mod:GetLocalizedStrings()

mod:SetRevision("20200611161557")
mod:SetCreatureID(15990)
mod:SetEncounterID(1114)
```

Reply:
`name` and `eId` is always provided by DBM (see: https://github.com/DeadlyBossMods/DBM-Classic/blob/master/DBM-Core/DBM-Core.lua#L11111) eId is set by SetEncounterID(...) which is in every DBM encounter file.

I've made LibEncounters based on this list https://wow.gamepedia.com/DungeonEncounterID. This list matches against the encounterID's DBM uses. LibEncounters does not use any Blizzard API calls, it only provides information that is not provided via Blizzard API in Classic. So the responsibility is with the addon using it to provide the correct `eId`. In this case it's EPGP-Classic, which uses the `eId` of DBM, which is identical to the DungeonEncounterID list. I've tested this in BWL on classic.

I chose to move to encounters, because I think it's a better term for what players are doing. I came to this conclusion when I was validating AQ40 and you have the "Twin Emperors" .. that's a fight with 2 NPC's, two creatureIds in DBM `mod:SetCreatureID(15276, 15275)` and fills `mod.combatInfo.name` differently. I actually think the addon in it's current state will bug out when you run Twin Emperors. That's why I decided to move to work with "Encounters" instead of "Boss"-es.